### DISCUSSION TITLE
`vim.treesitter.get_parser()` throws error for `typescriptreact` filetype

### DISCUSSION THREAD

**Discussion by numToStr** [nvim-treesitter/nvim-treesitter#4091] [Q&A]
Running `:lua vim.treesitter.get_parser()` in a `tsx` file throws error because parser for `typescriptreact` filetype is called `tsx` (and for `javascriptreact` is called `jsx`).

There are APIs in nvim-treesitter (discussed in https://github.com/nvim-treesitter/nvim-treesitter/issues/796) to tackle this issue but I wonder how this could be resolved by not having a dependency on nvim-treesitter and considering that this is not limited to `tsx` or `jsx`.

**Comment by clason** [comment]
Just use `vim.treesitter.get_parser(0, 'tsx')`; all that nvim-treesitter does is create a [simple table mapping filetypes to languages](https://github.com/nvim-treesitter/nvim-treesitter/blob/5afa9e50407c137c61082627a4456ab99cfbed18/lua/nvim-treesitter/parsers.lua#L4-L20) (for the few where these are not the same).
